### PR TITLE
add style and markers to map

### PR DESCRIPTION
**Pull request by nif33**
create styled map type using json
add markers using hard-coded locations
markers open an infowindow when clicked

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -16,9 +16,117 @@
...
+        "featureType": "water",
+        "elementType": "geometry.fill",
+        "stylers": [
+          {
+            "color": "#d3f0fa"
+          }
+        ]
+      }
+    ],
+  {name: 'Styled Map'});
```
Comment (chenzizhao) [inline portfolio/src/main/webapp/script.js]: NIT: Maybe store this big list as a separate mapStyle.json file and read from that local file when you need it? I think `parse()` should work nicely once you load the .json file but I haven't found a nice function to load local .json files.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -16,9 +16,117 @@
...
+  for(const location of locations){
+    // create DOM element for infowindow
+    const infoText = document.createElement('p');
+    const title = document.createElement('strong');
+    title.appendChild(document.createTextNode(location.name));
+    infoText.appendChild(title);
+    infoText.appendChild(document.createElement('br'));
+    infoText.appendChild(document.createTextNode(location.info));
+
+    const infowindow = new google.maps.InfoWindow({
```
Comment (chenzizhao) [inline portfolio/src/main/webapp/script.js]: NIT:  `infoWindow` camel case

**Review by chenzizhao — APPROVED** [review]
Logic LGTM. If you can find a nice way to load local .json files definitely let me know 😃  It is certainly not mandatory for the starter project though imo.

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (nif33) [reply portfolio/src/main/webapp/script.js]: ah I wanted to do this too and I found this https://codepen.io/KryptoniteDove/post/load-json-file-locally-using-pure-javascript but it does not look pretty haha. We can both try it!

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -16,9 +16,117 @@
  * Creates a map and adds it to the page
  */
```
Comment (juanalonsocruz) [inline portfolio/src/main/webapp/script.js:17]: Overall styleguide nit: Be consistent and use single quotes for all strings and object properties.

**Review by juanalonsocruz — APPROVED** [review]
It would be nice to find a way to organize API request settings into its own file or its own section. Not familiar with this. So approving anyways lol.

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (ajpaon) [reply portfolio/src/main/webapp/script.js]: Personally I think it's simpler to keep this in JS, but if you like the separation you can always move it to a separate map_styles.js file.

**Review by ajpaon — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (nif33) [reply portfolio/src/main/webapp/script.js]: ah ok, I was planning to separate the files anyways in a later PR, this file is getting way too long

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (nif33) [reply portfolio/src/main/webapp/script.js]: Done

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (nif33) [reply portfolio/src/main/webapp/script.js:17]: Done. I miss C++ where there's only option haha

**Pull request merged**